### PR TITLE
Integrate ShellCheck in linting script

### PR DESCRIPTION
**Pull request by tjjordan**

### CODE REVIEW

File: scripts/lint.sh [hunk 1]
```diff
@@ -94,4 +94,16 @@ if [ ! -z "$output" ]; then
 	exitcode=1
 fi
 
+# ShellCheck linting
+for file in `find . -name "*.sh"`
+do
+	if ! [[ $file == ./third_party/* ]] && ! [[ $file == ./cmake-* ]] && ! [[ $file == ./clang-* ]];
```
Comment (mweisgut) [inline scripts/lint.sh:100]: Is `! [[ $file == ./clang-* ]]` used to exclude clang build directories? Build directories can be named arbitrarily so they wouldn't be generally excluded in any cases.

How about including certain directories instead? For Hyrise, scripts are expected to be located in the scripts directory. In addition, `install_dependencies.sh` is located in the root directory.

How about only considering sh-files located in the project's root or `scripts` directory?

File: scripts/lint.sh [hunk 2]
```diff
@@ -94,4 +94,16 @@ if [ ! -z "$output" ]; then
 	exitcode=1
 fi
 
+# ShellCheck linting
+for file in `find . -name "*.sh"`
```
Comment (mweisgut) [inline scripts/lint.sh:98]: For consistency reasons, how about using the `$(<bash command>)` syntax as used above, see line 68.

Comment (Bouncner) [reply scripts/lint.sh:98]: As far as I know, backticks are considered "old style" and `$(..)` should be preferred.

Comment (tjjordan) [reply scripts/lint.sh:98]: Thanks for pointing this out! :)

File: scripts/lint.sh (hunk 1 above)
Comment (tjjordan) [reply scripts/lint.sh:100]: This is the better way, done!

Comment (Bouncner) [reply scripts/lint.sh:101]: When you're only accessing sh-files in the scripts directory, you can remove all the checks as Marcel hinted.

Comment (mweisgut) [reply scripts/lint.sh:101]: In addition, I would include `install_dependencies.sh` (or .sh files located in the root directory in general).

**Comment by Bouncner** [comment]
What is the status here?

**Comment by tjjordan** [comment]
It has not been forgotten.
The SpellCheck errors in the scripts need to be fixed be corrected otherwise the CI will fail.
I'm planning to do this after the PartialIndex-PR or when I've some spare time.

**Comment by dey4ss** [comment]
Still did not get if it's S<ins>p</ins>ellCheck or S<ins>h</ins>ellcheck (and at that point, I'm too afraid to ask).

**Comment by mweisgut** [comment]
Closing it for now. Feel free to reopen if it becomes a priority.

**Pull request closed without merging**